frontend/src/components/OrderPage.js

```diff
@@ -113,6 +122,47 @@ class OrderPage extends Component {
     this.getOrderDetails(this.state.orderId);
   }
 
+  handleWebln = async (data) => {
+    const webln = await getWebln();
+    // If Webln implements locked payments compatibility, this logic might be simplier
+    if (data.is_maker & data.status == 0) {
+      webln.sendPayment(data.bond_invoice);
+      this.setState({ waitingWebln: true, openWeblnDialog: true});
+    } else if (data.is_taker & data.status == 3) {
+      webln.sendPayment(data.bond_invoice);
+      this.setState({ waitingWebln: true, openWeblnDialog: true});
+    } else if (data.is_seller & (data.status == 6 || data.status == 7 )) {
+      webln.sendPayment(data.escrow_invoice);
+      this.setState({ waitingWebln: true, openWeblnDialog: true});
+    } else if (data.is_buyer & (data.status == 6 || data.status == 8 )) {
+      webln.makeInvoice(data.trade_satoshis)
+        .then((invoice) => {
+          if (invoice) {
+            this.sendWeblnInvoice(invoice.paymentRequest);
+            this.setState({ waitingWebln: true, openWeblnDialog: true});
+          }
+        }).catch(() => {
+          this.setState({ waitingWebln: false, openWeblnDialog: false });
```

Comment:
If the invoice is successfully submitted, the dialog with the spinner stays (one can see in the background that the status of the order advances). Only closes the Dialog when the Alby window is Canceled.

Reply:
ON that case we need to set some special conditions here, so the dialog is closed automatically with invoices but keeps open with payment, makes sense.